packages/mini-browser/src/browser/mini-browser-content.ts

```diff
@@ -511,8 +515,16 @@ export class MiniBrowserContent extends BaseWidget {
         return element;
     }
 
-    protected mapLocation(location: string): Promise<string> {
-        return this.locationMapper.map(location);
+    protected async mapLocation(location: string): Promise<string> {
+        const [hostPattern, mappedLocation] = await Promise.all([
+            this.miniBrowserEnvironment.hostPattern,
+            this.locationMapper.map(location),
+        ]);
+        // The `LocationMapper` will most likely resolve something on Theia's main host.
+        // But we need to re-route those to the virtual host where the mini-browser accepts requests.
+        const url = new URL(mappedLocation);
+        url.host = hostPattern.replace('{{hostname}}', url.host);
```

Comment:
Does that mean any URL will be resolved to something like `mini-browser.*`?
Let's say, I'm trying to navigate to `https://www.google.com/`, but eventually, it will be re-routed to
`https://www.mini-browser.google.com/`.

Should we re-route only those URLs that are referring to one the Theia's main hosts (`THEIA_HOSTS`)?

Reply:
it is a bug and should be reverted. It should be done like in https://github.com/eclipse-theia/theia/pull/8759#discussion_r527523823

Reply:
Placed the logic in the wrong place, I should only change the `MiniBrowserEndpoint` endpoint. I had misunderstood how this method worked.